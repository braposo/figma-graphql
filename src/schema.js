const { makeExecutableSchema } = require("graphql-tools");
const { GraphQLDateTime } = require("graphql-iso-date");
const mergeSchema = require("./types");

const typeDefs = [
    `
    scalar DateTime

    type Query {
        version: String!
    }

    type Mutation {
        noop: String
    }
    
    type Subscription {
        noop: String
    }
    
    enum BlendMode {
        PASS_THROUGH
        NORMAL
        DARKEN
        MULTIPLY
        LINEAR_BURN
        COLOR_BURN
        LIGHTEN
        SCREEN
        LINEAR_DODGE
        COLOR_DODGE
        OVERLAY
        SOFT_LIGHT
        HARD_LIGHT
        DIFFERENCE
        EXCLUSION
        HUE
        SATURATION
        COLOR
        LUMINOSITY
    }
    
    enum StrokeAlign {
        INSIDE
        OUTSIDE
        CENTER
    }
`,
];

const resolvers = {
    DateTime: GraphQLDateTime,
    Query: {
        version: () => "1.4",
    },
};

module.exports = {
    schema: makeExecutableSchema(mergeSchema({ typeDefs, resolvers })),
    context: () => ({}),
};
