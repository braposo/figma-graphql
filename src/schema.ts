import { makeExecutableSchema } from "graphql-tools";
import { GraphQLDateTime } from "graphql-iso-date";
import mergeSchema from "./types";

const typeDefs = [
    `
    scalar DateTime

    type Query {
        version: String!
    }

    type Mutation {
        noop: String
    }
    
    type Subscription {
        noop: String
    }
`,
];

const resolvers = {
    DateTime: GraphQLDateTime,
    Query: {
        version: () => "1.4",
    },
};

export default makeExecutableSchema(mergeSchema({ typeDefs, resolvers }));
