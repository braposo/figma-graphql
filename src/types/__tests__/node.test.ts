/* eslint-env jest */

import { graphql } from "graphql";
import schema from "../../schema";

const figmaFile = "cLp23bR627jcuNSoBGkhL04E";

jest.setTimeout(10000);

describe("Node", () => {
    test("returns default export image", async () => {
        const query = `
            query {
                file(id: "${figmaFile}") {
                    rectangles {
                        export
                    }
                }
            }
        `;

        const response = await graphql(schema, query, null, { fileId: figmaFile });
        expect(response.data).not.toBeUndefined();

        const rectangles = response.data && response.data.file.rectangles;
        rectangles.forEach(rectangle => {
            const { export: nodeExport } = rectangle;
            expect(nodeExport).not.toBeNull();
        });
    });

    test("returns jpg export", async () => {
        const query = `
            query {
                file(id: "${figmaFile}") {
                    rectangles {
                        export(params: { format: jpg })
                    }
                }
            }
        `;

        const response = await graphql(schema, query, null, { fileId: figmaFile });
        expect(response.data).not.toBeUndefined();

        const rectangles = response.data && response.data.file.rectangles;
        rectangles.forEach(rectangle => {
            const { export: nodeExport } = rectangle;
            expect(nodeExport).not.toBeNull();
        });
    });

    test("returns css export", async () => {
        const query = `
            query {
                file(id: "${figmaFile}") {
                    rectangles {
                        export(params: { format: css })
                    }
                }
            }
        `;

        const response = await graphql(schema, query, null, { fileId: figmaFile });
        expect(response.data).not.toBeUndefined();

        const rectangles = response.data && response.data.file.rectangles;
        rectangles.forEach(rectangle => {
            const { export: nodeExport } = rectangle;
            expect(nodeExport).toMatchSnapshot();
        });
    });
});
