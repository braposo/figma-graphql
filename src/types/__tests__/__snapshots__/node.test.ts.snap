// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node returns css export 1`] = `"width: 201px; height: 109px;"`;

exports[`Node returns css export 2`] = `"width: 201px; height: 109px;"`;

exports[`Node returns css export 3`] = `"width: 201px; height: 109px;"`;

exports[`Node returns css export 4`] = `"width: 144px; height: 62px;"`;
